lib/features/children/edit_child/pages/edit_child_page.dart

```diff
@@ -0,0 +1,234 @@
+    final currency = NumberFormat.simpleCurrency(
+      name: Util.getCurrencyName(country: Country.fromCode(user.country)),
+    );
+
+    return BlocConsumer<EditChildCubit, EditChildState>(
+      listener: (context, state) {
+        log('edit child state changed on $state');
+        if (state is EditChildExternalErrorState) {
+          ScaffoldMessenger.of(context).showSnackBar(
+            SnackBar(
+              content: Text(
+                state.errorMessage,
+                textAlign: TextAlign.center,
+              ),
+              backgroundColor: Theme.of(context).errorColor,
+            ),
+          );
+        } else if (state is EditChildInputState ||
+            state is EditChildInputErrorState) {
+          _updateInputFields(
+            state.profileDetails,
+            state.child,
+            currency.currencySymbol,
+          );
+        } else if (state is EditChildSuccessState) {
+          context
+              .read<ChildDetailsCubit>()
+              .fetchChildDetails(state.profileDetails.profile);
+          context.pop();
+        }
+      },
+      builder: (context, state) {
+        return Scaffold(
+          appBar: AppBar(
+            automaticallyImplyLeading: false,
+            actions: [
+              Expanded(
+                child: Container(
+                  alignment: Alignment.centerLeft,
+                  child: TextButton.icon(
+                    icon: const Icon(Icons.arrow_back_ios_new_sharp),
+                    label: Text(
+                      locals.cancel,
+                      style: Theme.of(context).textTheme.titleMedium?.copyWith(
+                            fontWeight: FontWeight.bold,
+                            color: AppTheme.inputFieldBorderSelected,
+                          ),
+                    ),
+                    onPressed: () {
+                      context.pop();
+                      AnalyticsHelper.logEvent(
+                        eventName: AmplitudeEvents.backClicked,
+                      );
+                    },
+                  ),
+                ),
+              ),
+              Expanded(
+                child: Center(
+                  child: Text(
+                    //TODO: POEditor
+                    'Edit Profile',
+                    style: Theme.of(context).textTheme.titleLarge?.copyWith(
+                          fontWeight: FontWeight.bold,
+                        ),
+                  ),
+                ),
+              ),
+              const Spacer(),
+            ],
+          ),
+          body: state is EditChildUploadingState
+              ? const Center(
+                  child: CircularProgressIndicator(
+                    color: AppTheme.sliderIndicatorFilled,
+                  ),
+                )
+              : state is EditChildInputState ||
+                      state is EditChildInputErrorState
+                  ? SizedBox(
```

Comment:
Yeah.... still not a big fan of this short hand conditionals. especially here you have a more complex one